src/web/rustdoc.rs

```diff
@@ -483,6 +484,24 @@ pub fn rustdoc_html_server_handler(req: &mut Request) -> IronResult<Response> {
 
     let latest_path = format!("/crate/{}/latest{}{}", name, target_redirect, query_string);
 
+    let canonical_url = if krate.documentation_url.is_none()
+        || krate
+            .documentation_url
+            .as_ref()
+            .unwrap()
+            .starts_with("https://docs.rs/")
+    {
+        format!(
+            "{}://{}/{}/latest/{}",
+            req.url.scheme(),
+            req.url.host(),
+            name,
+            inner_path
+        )
+    } else {
+        "".to_string()
```

Comment:
shouldn't this be the crate's `documentation_url`?
cc @Nemo157 I know you had concerns about this not actually being a rustdoc page

Reply:
since the canonical url should point to the _same page_, I don't see how this would work for `documentation_url` where we don't even know if it's generated with rustdoc, or just some mdbook.